### PR TITLE
fix(FEC-12132): Add documentation

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Add documentation and update types

solves: FEC-12132

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [x] Docs have been updated

### CODE REVIEW

File: docs/guide.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+### First step: setup
+
+First include `playkit-ui-managers.js` after kaltura-player script in your web page.
+
+```html
+  <script src="https://raw.githack.com/kaltura/kaltura-player-js/master/dist/kaltura-ovp-player.js"></script>
+  <script src="./playkit-ui-managers.js"></script>
+```
+
+Add the ui-managers to the setup config like any other plugin in addition to the plugin uses the ui-managers.
```
Comment (SivanA-Kaltura) [inline docs/guide.md:19]:
```suggestion
Add the ui-managers plugin configuration object to the player plugins configuration.
```

File: docs/guide.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        uiManagers: {}
+      }
+    };
+
+const player = KalturaPlayer.setup(config);
+```
+
+### Second step: access and use the desired service inside the plugin
+
+get and use the desired service using the player's api inside the loadMedia() hook of your plugin
```
Comment (SivanA-Kaltura) [inline docs/guide.md:38]:
```suggestion
You can access and use the desired service by using the player's api inside the loadMedia() hook of your plugin.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+[kaltura player]: https://github.com/kaltura/kaltura-player-js.
+[ecmascript6]: https://github.com/ericdouglas/ES6-Learning#articles--tutorials
+[typescript]: https://www.typescriptlang.org/
+[typescript compiler]: https://www.typescriptlang.org/docs/handbook/compiler-options.html
+[webpack]: https://webpack.js.org/
+
+## services
+
+- ###Side Panels
```
Comment (SivanA-Kaltura) [inline README.md:29]:
```suggestion
- Side Panels
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  manages and controls the side panel items\
```
Comment (SivanA-Kaltura) [inline README.md:30]:
```suggestion
  Manages and controls the side panel items, including:\
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  \
+  that's includes:
```
Comment (SivanA-Kaltura) [inline README.md:32]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  - When an item is activated in the panel, the other items are automatically deactivated in that panel.
+  - Automatically deactivates a panel when its counter panel is activated\
```
Comment (SivanA-Kaltura) [inline README.md:34]:
```suggestion
  - When a panel is activated, its counter panel is automatically deactivated.\
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  \
+    Full API documentation can be found [here](https://kaltura.github.io/playkit-js-ui-managers/docs/api/classes/side_panels_manager.SidePanelsManager.html)
+  
+- ###Upper bar manager
```
Comment (SivanA-Kaltura) [inline README.md:38]:
```suggestion
- Upper bar manager
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+  manages and controls the upper bar menu\
```
Comment (SivanA-Kaltura) [inline README.md:39]:
```suggestion
  Manages and controls the upper bar menu, including:\
```

File: README.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  \
+  that's includes:
```
Comment (SivanA-Kaltura) [inline README.md:41]:
```suggestion
```

File: README.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+  - sets the order icons are shown.
```
Comment (SivanA-Kaltura) [inline README.md:42]:
```suggestion
  - Setting the order in which icons are shown.
```

File: README.md [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+  - Increases and decreases the number of icons that appear depending on the size of the player\
```
Comment (SivanA-Kaltura) [inline README.md:43]:
```suggestion
  - Increasing and decreasing the number of icons that appear depending on the size of the player.\
```

File: README.md [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+[typescript compiler]: https://www.typescriptlang.org/docs/handbook/compiler-options.html
+[webpack]: https://webpack.js.org/
+
+## services
+
+- ###Side Panels
+  manages and controls the side panel items\
+  \
+  that's includes:
+  - When an item is activated in the panel, the other items are automatically deactivated in that panel.
```
Comment (SivanA-Kaltura) [inline README.md:33]:
```suggestion
  - When an item is activated in a panel, the other items in that panel are automatically deactivated.
```

File: README.md [hunk 13]
```diff
@@ -0,0 +1,153 @@
+[![Build Status](https://app.travis-ci.com/kaltura/playkit-js-ui-managers.svg?branch=master)](https://app.travis-ci.com/kaltura/playkit-js-ui-managers)
+[![](https://img.shields.io/npm/v/@playkit-js/ui-managers/latest.svg)](https://www.npmjs.com/package/@playkit-js/ui-managers)
+[![](https://img.shields.io/npm/v/@playkit-js/ui-managers/canary.svg)](https://www.npmjs.com/package/@playkit-js/ui-managers/v/canary)
+
+# playkit-js-ui-managers
+
+playkit-js-ui-managers is a [kaltura player] plugin that groups several UI services,
+in order to simplify and facilitate the customization of the [kaltura player] UI by providing a simple and clean API.
+
+Each service manages a different area of UI functionality
```
Comment (SivanA-Kaltura) [inline README.md:10]:
```suggestion
Each service manages a different area of UI functionality.
```

File: docs/guide.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
+# usage guide
```
Comment (SivanA-Kaltura) [inline docs/guide.md:1]:
```suggestion
# Usage guide
```

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: docs/guide.md (hunk 1 above)
Comment (JonathanTGold) [reply docs/guide.md:19]: fixed

File: README.md (hunk 4 above)
Comment (JonathanTGold) [reply README.md:30]: fix

File: README.md (hunk 5 above)
Comment (JonathanTGold) [reply README.md:32]: fixed

File: README.md (hunk 11 above)
Comment (JonathanTGold) [reply README.md:43]: fixed

File: README.md (hunk 12 above)
Comment (JonathanTGold) [reply README.md:33]: fixed

File: README.md (hunk 13 above)
Comment (JonathanTGold) [reply README.md:10]: fixed

File: docs/guide.md (hunk 14 above)
Comment (JonathanTGold) [reply docs/guide.md:1]: fixed

File: README.md (hunk 3 above)
Comment (SivanA-Kaltura) [reply README.md:27]: a space is missing between the number signs and the text

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**